### ISSUE TITLE
Will Windows support be added, if so, when?

### ISSUE THREAD

**Issue by Porfik** [darglein/ADOP#6]

**Comment by andybak** [comment]
Or failing that - I'd be interested to know if it's possible to run this under WSL with the recent support for direct GPU access.

**Comment by darglein** [comment]
I'm currently on an internship without access to a suitable windows machine. I will try it once I'm back in January.

**Comment by Ilyaprok** [comment]
> Or failing that - I'd be interested to know if it's possible to run this under WSL with the recent support for direct GPU access.

Hello! I have tried in WSL2 with X server GUI, but unsuccessed.
There is some render with 3D view. 
![Безымянный2](https://user-images.githubusercontent.com/31244872/140500166-63850b86-f89c-4dd8-8c50-38a7fc524020.png)
But neuro render "F2" closes app.
`Assertion 'cudaGraphicsGLRegisterImage(&graphic_resource, gl_buffer, gl_target, cudaGraphicsRegisterFlagsSurfaceLoadStore) == cudaSuccess' failed!
  File: /home/robot/neuro_render/ADOP/External/saiga/src/saiga/cuda/interop.h:72
  Function: void Saiga::CUDA::Interop::initImage(unsigned int, gl::GLenum)
  Message: OS call failed or operation not supported on this OS
Aborted`

**Comment by darglein** [comment]
Does the libtorch/cuda stuff work on WSL? Can you try executing the `test_loss` and `adop_train` please?

If this works I can add an option to compile the viewer without CUDA/GL interop. This might reduce performance a little bit, but at least it displays an image.

**Comment by robsingh9** [comment]
I tried on WSL2 for Windows 10 and could not get it working. The compile works fine until make -j10 where nothing happens. Works great on Ubuntu however.

**Comment by metya** [comment]
I tried on WSL2 too.
And it is fully compilable, but doesn't work and doesn't render anything because of an error with syscalls in windows I think.

So I tried it in my Arch and it is doesn't compile at all.

So even if it is on Linux, it is doesn't working((

**Comment by darglein** [comment]
@metya What is the issue on Arch? Did you follow the install scripts?

**Comment by blurgyy** [comment]
> So I tried it in my Arch and it is doesn't compile at all.
> 
> 
> 
> So even if it is on Linux, it is doesn't working((

That is not true, I have managed to compile on latest Arch after some tinkering.  Some notable changes include:

1. Install `cuda-11.1` and `gcc-9` on your system (because in the AUR there is no `cuda-11.2` and I find `cuda-11.1` to work just fine with `cudatoolkit=11.2`), and prepend `/opt/cuda-11.1` to your `CUDA_PATH` and some relevant paths to your `PATH` (refer to your `/etc/profile.d/cusa.sh`)
2. Install `freeimage` in your conda environment (this seems to be added since commit ec1432e)
3. Compile `srt` with the same environment as PyTorch and ADOP, otherwise compilation of ADOP fails at the final linking stage.  You could first add https://github.com/Haivision/srt.git to submodules, then create a `install_srt.sh` similar to the `install_pytorch.sh` provided in the repo.

I'm not here to give support, but you should certainly try harder, especially when you decided to use Arch.

**Comment by darglein** [comment]
@blurgyy Thanks for the short Arch guide. May I ask why you need a system-wide install of CUDA? Shouldn't the cuda toolkit which is installed in the conda environment  be enough?

**Comment by blurgyy** [comment]
> May I ask why you need a system-wide install of CUDA? Shouldn't the cuda toolkit which is installed in the conda environment be enough?

Oh I think I just tried to match the environment given in the repo in the first place, did not try to build with latest CUDA (11.5 on Arch).  You are probably right!  (Also, I just checked that I also created my environment with `cudatoolkit=11.1` to match a system CUDA version 11.1)

**Comment by metya** [comment]
@darglein Yes. I completely follow the install instructions and scripts row by row.
since it is not very related to this issue, I create a new one.

I will try instruction from @blurgyy and if it doesn't help, I'll create a new issue and will link it here.

@blurgyy thanks! I will try it! 
I've installed freeimage in my conda env.
But I've realized that I have cuda-11.4.2-1 in my arch. 
And didn't compile srt by own.

**Comment by metya** [comment]
So I tried what said @blurgyy and it didn't work.
I describe detail here #27

**Comment by stjema** [comment]
i can share my screen. i have win11, we can see if it works
